common/src/main/java/org/opensearch/ml/common/connector/ConnectorAction.java

```diff
@@ -183,6 +186,24 @@ public static ConnectorAction parse(XContentParser parser) throws IOException {
 
     public enum ActionType {
         PREDICT,
-        EXECUTE
+        EXECUTE,
+        BATCH;
```

Comment:
`BATCH` isn't quite intuitive. In the long run, if we need BATCH Execute, what would we name it then? May be `BATCH_PREDICT`?

Reply:
Is batch execute a real use case? Currently the batch can be used to map to different method for different servers platforms, like batch-transform in SageMaker, batch-predict in bedrock, batches in OpenAI, and embed-jobs for cohere.

Reply:
well `EXECUTE` wasn't a real use case until we introduced agent. So similarly I don't think there's any point to cancel out any possibility. In addition, I don't think we want to face the same issue like `REMOTE`, where we don't know what type of connectors people are using, all we know is `REMOTE`. Is this possible to reflect what lesson we learned?

Reply:
or May be `BATCH_INFERENCE`?

Reply:
BATCH_PREDICT is fine. This is better consistent with PREDICT. The "REMOTE" is a different problem, we cannot enumerate all model types as they are defined by customers. Also, the more different types you added, the more complicate interphase you will introduce for customers to config models. There are always methods for us to summaries the different types of models by checking the connector url/processingfunctions, etc. Using "BATCH_PREDICT" would also require customers to type one more word in the APIs like <url path>/_batch_predict.

Reply:
>Using "BATCH_PREDICT" would also require customers to type one more word in the APIs like /_batch_predict.

That's not a big deal. I don't think customers are going to manually invoke this api everytime.